### PR TITLE
Nav graph

### LINKED ISSUE #398: Adapt Navhost for Navigation in VigyBag App
## I will make the navigation using Navigation graphs for easy navigation.
**Using Official references as suggested:-**
 _https://developer.android.com/guide/navigation/get-started#add-navhost_

### PR DESCRIPTION
**Pull request by essy16**
Fixes issue #398

### CODE REVIEW

File: Android/app/src/main/java/com/vigbag/android/MainActivity.kt [hunk 1]
```diff
@@ -12,13 +12,13 @@ class MainActivity : AppCompatActivity() {
         setTheme(R.style.Theme_VigyBag)
         binding=ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        replaceFragment(SignInFragment())
+//        replaceFragment(SignInFragment())
```
Comment (07jasjeet) [inline Android/app/src/main/java/com/vigbag/android/MainActivity.kt]: Please remove all the commented out code :)

File: Android/app/src/main/java/com/vigbag/android/MainActivity.kt [hunk 2]
```diff
@@ -12,13 +12,13 @@ class MainActivity : AppCompatActivity() {
         setTheme(R.style.Theme_VigyBag)
         binding=ActivityMainBinding.inflate(layoutInflater)
```
Comment (07jasjeet) [inline Android/app/src/main/java/com/vigbag/android/MainActivity.kt]:
```suggestion
```

File: Android/app/src/main/java/com/vigbag/android/MainActivity.kt [hunk 3]
```diff
@@ -12,13 +12,13 @@ class MainActivity : AppCompatActivity() {
         setTheme(R.style.Theme_VigyBag)
```
Comment (07jasjeet) [inline Android/app/src/main/java/com/vigbag/android/MainActivity.kt]:
```suggestion
```

File: Android/app/src/main/java/com/vigbag/android/MainActivity.kt (hunk 2 above)
Comment (07jasjeet) [inline Android/app/src/main/java/com/vigbag/android/MainActivity.kt]:
```suggestion
        setContentView(R.layout.activity_main)
```

Comment (07jasjeet) [reply Android/app/src/main/java/com/vigbag/android/MainActivity.kt]: We do not need binding as Fragment container view automatically sets up navigation for us.

File: Android/app/src/main/java/com/vigbag/android/MainActivity.kt (hunk 3 above)
Comment (07jasjeet) [reply Android/app/src/main/java/com/vigbag/android/MainActivity.kt]: We shouldn't need this. Please test and remove.

File: Android/app/src/main/java/com/vigbag/android/MainActivity.kt (hunk 2 above)
Comment (07jasjeet) [reply Android/app/src/main/java/com/vigbag/android/MainActivity.kt]: Also remove binding variable declaration on line 9

File: Android/build.gradle [hunk 4]
```diff
@@ -1,6 +1,6 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id 'com.android.application' version '8.0.2' apply false
-    id 'com.android.library' version '8.0.2' apply false
+    id 'com.android.application' version '7.4.2' apply false
```
Comment (07jasjeet) [inline Android/build.gradle:3]: Why downgrade this?

**Review by 07jasjeet — CHANGES_REQUESTED** [review]
Nice work. A few changes and we should be good.

File: Android/build.gradle (hunk 4 above)
Comment (essy16) [reply Android/build.gradle:3]: This one is compatible with my android studio

File: Android/app/src/main/java/com/vigbag/android/MainActivity.kt (hunk 2 above)
Comment (essy16) [reply Android/app/src/main/java/com/vigbag/android/MainActivity.kt]: Ooh

File: Android/build.gradle (hunk 4 above)
Comment (07jasjeet) [reply Android/build.gradle:3]: Revert this for now and keep these changes on your local. Send the error you're facing and I'll try to fix it.

File: Android/app/src/main/java/com/vigbag/android/SignInFragment.kt [hunk 5]
```diff
@@ -40,9 +41,10 @@ class SignInFragment : Fragment(R.layout.fragment_sign_in) {
         _binding = FragmentSignInBinding.bind(view)
         
         binding.txtSignup.setOnClickListener {
-            parentFragmentManager.beginTransaction()
-                .replace(R.id.frameLayout, SignUpFragment())
-                .commit()
+            val navController=Navigation.findNavController(requireView())
+            navController.navigate(R.id.action_signInFragment_to_signUpFragment)
```
[the remark this answers is not in the document]
Comment (07jasjeet) [reply Android/app/src/main/java/com/vigbag/android/SignInFragment.kt:45]: @essy16 Fix this so we can move forward.

File: Android/build.gradle (hunk 4 above)
Comment (essy16) [reply Android/build.gradle:3]: What do you mean by revert?

Comment (07jasjeet) [reply Android/build.gradle:3]: Nvm leave it just complete rest of the changes

Comment (essy16) [reply Android/build.gradle:3]: @07jasjeet I made all the other changes

Comment (07jasjeet) [reply Android/build.gradle:3]: @essy16 please recheck there are still comments to be resolved above.

Comment (essy16) [reply Android/build.gradle:3]: @07jasjeet i made the recent commit and I did all those ...please check

Comment (essy16) [reply Android/build.gradle:3]: @07jasjeet i made the recent commit and I did all those ...please check

File: Android/app/src/main/java/com/vigbag/android/SignInFragment.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (07jasjeet) [reply Android/app/src/main/java/com/vigbag/android/SignInFragment.kt:45]: @essy16 fix this

File: Android/app/src/main/java/com/vigbag/android/SignUpFragment.kt [hunk 6]
```diff
@@ -29,9 +30,9 @@ class SignUpFragment : Fragment(R.layout.fragment_sign_up) {
         _binding = FragmentSignUpBinding.bind(view)
         
         binding.txtLogin.setOnClickListener {
-            parentFragmentManager.beginTransaction()
-                .replace(R.id.frameLayout, SignInFragment())
-                .commit()
+            val navController= Navigation.findNavController(requireView())
+            navController.navigate(R.id.action_signUpFragment_to_signInFragment)
```
[the remark this answers is not in the document]
Comment (07jasjeet) [reply Android/app/src/main/java/com/vigbag/android/SignUpFragment.kt:34]: @essy16 fix this

File: Android/app/src/main/java/com/vigbag/android/SignInFragment.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (essy16) [reply Android/app/src/main/java/com/vigbag/android/SignInFragment.kt:45]: It's not on my code

Comment (07jasjeet) [inline Android/app/src/main/java/com/vigbag/android/SignInFragment.kt]:
```suggestion
            findNavController().navigate(R.id.action_signInFragment_to_signUpFragment)
```

File: Android/app/src/main/java/com/vigbag/android/SignInFragment.kt [hunk 7]
```diff
@@ -40,9 +41,10 @@ class SignInFragment : Fragment(R.layout.fragment_sign_in) {
         _binding = FragmentSignInBinding.bind(view)
         
         binding.txtSignup.setOnClickListener {
-            parentFragmentManager.beginTransaction()
-                .replace(R.id.frameLayout, SignUpFragment())
-                .commit()
+            val navController=Navigation.findNavController(requireView())
```
Comment (07jasjeet) [inline Android/app/src/main/java/com/vigbag/android/SignInFragment.kt]:
```suggestion
```

File: Android/app/src/main/java/com/vigbag/android/SignUpFragment.kt (hunk 6 above)
Comment (07jasjeet) [inline Android/app/src/main/java/com/vigbag/android/SignUpFragment.kt]:
```suggestion
            findNavController().navigate(R.id.action_signUpFragment_to_signInFragment)
```

File: Android/app/src/main/java/com/vigbag/android/SignUpFragment.kt [hunk 8]
```diff
@@ -29,9 +30,9 @@ class SignUpFragment : Fragment(R.layout.fragment_sign_up) {
         _binding = FragmentSignUpBinding.bind(view)
         
         binding.txtLogin.setOnClickListener {
-            parentFragmentManager.beginTransaction()
-                .replace(R.id.frameLayout, SignInFragment())
-                .commit()
+            val navController= Navigation.findNavController(requireView())
```
Comment (07jasjeet) [inline Android/app/src/main/java/com/vigbag/android/SignUpFragment.kt]:
```suggestion
```

**Review by 07jasjeet — CHANGES_REQUESTED** [review]

File: Android/app/src/main/java/com/vigbag/android/MainActivity.kt [hunk 9]
```diff
@@ -9,16 +9,9 @@ class MainActivity : AppCompatActivity() {
     lateinit var binding:ActivityMainBinding
```
Comment (07jasjeet) [inline Android/app/src/main/java/com/vigbag/android/MainActivity.kt]:
```suggestion
```

**Comment by 07jasjeet** [comment]
@essy16 I have fixed the code myself for now but to state it, this is all about changing other's code. Saying "it's not my code" is not the way to go. For now the changes look good to me. Nice work!

**Comment by essy16** [comment]
I am sorry but the code was not in my code so I was wondering but I'm really sorry if you're offended

**Pull request merged**